File: benchmark/lua/sieve.lua
local limit = 10000
local sieve = {}
for i = 0, limit do sieve[i] = true end
sieve[0] = false
sieve[1] = false
local start = os.clock()
for p = 2, math.floor(math.sqrt(limit)) do
    if sieve[p] then
        for multiple = p * p, limit, p do
            sieve[multiple] = false
        end
    end
end
local count = 0
for i = 2, limit do
    if sieve[i] then count = count + 1 end
end
local finish = os.clock()
print("Primes up to 10000: " .. count)
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmark/perl/sieve.pl
use Time::HiRes qw(time);
my $limit = 10000;
my @sieve = (1) x ($limit + 1);
$sieve[0] = 0; $sieve[1] = 0;
my $start = time();
for my $p (2..int(sqrt($limit))) {
    if ($sieve[$p]) {
        for (my $m = $p * $p; $m <= $limit; $m += $p) {
            $sieve[$m] = 0;
        }
    }
}
my $count = 0;
for my $i (2..$limit) {
    $count++ if $sieve[$i];
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "Primes up to 10000: $count
";
printf "Time: %.0fms
", $ms;
